pool/web/handler/public_login.ml

```diff
@@ -170,14 +207,21 @@ let request_reset_password_post req =
                ||> fun Contact.{ language; _ } ->
                CCOption.value ~default:context.Pool_context.language language)
         in
-        PasswordReset.create database_label message_language (Tenant tenant) user
+        make_message message_language user
         >== handle ~tags
         |>> Pool_event.handle_events ~tags database_label context.Pool_context.user
     in
-    redirect_to_with_actions
-      redirect_path
-      [ Message.set ~success:[ Pool_message.Success.PasswordResetSuccessMessage ] ]
-    >|> Lwt_result.return
+    let redirect () =
+      redirect_to_with_actions
+        redirect_path
+        [ Message.set ~success:[ Pool_message.Success.PasswordResetSuccessMessage ] ]
+      >|> Lwt_result.return
+    in
+    handle ()
+    ||> CCResult.get_lazy (fun err ->
+      let (_ : Pool_message.Error.t) = Pool_common.Utils.with_log_error err in
+      ())
+    >|> redirect
```

Comment:
Nearly the same mechanism is run twice behind each other. First here in the handler and then within `extract_happy_path_with_actions` a second time.

Let's discuss this tomorrow.